coretex/configuration/node.py

```diff
@@ -0,0 +1,376 @@
+        return self.getOptValue("nodeSecret", str)
+
+    @nodeSecret.setter
+    def nodeSecret(self, value: Optional[str]) -> None:
+        self._raw["nodeSecret"] = value
+
+    @property
+    def initScript(self) -> Optional[str]:
+        return self.getOptValue("initScript", str)
+
+    @initScript.setter
+    def initScript(self, value: Optional[str]) -> None:
+        self._raw["initScript"] = value
+
+    @property
+    def modelId(self) -> Optional[int]:
+        return self.getOptValue("modelId", int)
+
+    @modelId.setter
+    def modelId(self, value: Optional[int]) -> None:
+        self._raw["modelId"] = value
+
+    @property
+    def nearWalletId(self) -> Optional[str]:
+        return self.getOptValue("nearWalletId", str)
+
+    @modelId.setter
+    def nearWalletId(self, value: Optional[str]) -> None:
+        self._raw["nearWalletId"] = value
+
+    @property
+    def endpointInvocationPrice(self) -> Optional[float]:
+        return self.getOptValue("endpointInvocationPrice", float)
+
+    @modelId.setter
+    def endpointInvocationPrice(self, value: Optional[float]) -> None:
+        self._raw["endpointInvocationPrice"] = value
+
+    def validateRamField(self, ramLimit: int) -> Tuple[bool, int, str]:
+        isValid = True
+        message = ""
+
+        if ramLimit < config_defaults.MINIMUM_RAM:
+            isValid = False
+            raise DockerConfigurationException(
+                f"Minimum Node RAM requirement ({config_defaults.MINIMUM_RAM}GB) "
+                f"is higher than your current Docker desktop RAM limit ({ramLimit}GB). "
+                "Please adjust resource limitations in Docker Desktop settings to match Node requirements."
+            )
+
+        defaultRamValue = int(min(max(config_defaults.MINIMUM_RAM, ramLimit), config_defaults.DEFAULT_RAM))
+
+        if not isinstance(self._raw.get("nodeRam"), int):
+            isValid = False
+            message = (
+                f"Invalid config \"nodeRam\" field type \"{type(self._raw.get('nodeRam'))}\". Expected: \"int\""
+                f"Using default value of {defaultRamValue} GB"
+            )
+
+        if self.nodeRam < config_defaults.MINIMUM_RAM:
+            isValid = False
+            message = (
+                f"WARNING: Minimum Node RAM requirement ({config_defaults.MINIMUM_RAM}GB) "
+                f"is higher than the configured value ({self._raw.get('nodeRam')}GB)"
+                f"Overriding \"nodeRam\" field to match node RAM requirements."
+            )
+
+        if self.nodeRam > ramLimit:
+            isValid = False
+            message = (
+                f"WARNING: RAM limit in Docker Desktop ({ramLimit}GB) "
+                f"is lower than the configured value ({self._raw.get('nodeRam')}GB)"
+                f"Overriding \"nodeRam\" field to limit in Docker Desktop."
+            )
+
+        return isValid, defaultRamValue, message
+
+    def validateCPUCount(self, cpuLimit: int) -> Tuple[bool, int, str]:
```

Comment:
Change these 3 (RAM, Swap, and CPU) validation functions to return `Optional[Tuple[$ValueType, str]]`